src/app/tests/TestAttributePathParams.cpp

```diff
@@ -0,0 +1,104 @@
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+/**
+ *    @file
+ *      This file implements unit tests for CommandPathParams
+ *
+ */
+
+#include <app/AttributePathParams.h>
+#include <nlunit-test.h>
+#include <support/UnitTestRegistration.h>
+
+namespace chip {
+namespace app {
+namespace TestAttributePathParams {
+void TestSamePath(nlTestSuite * apSuite, void * apContext)
+{
+    AttributePathParams attributePathParams1(1, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    AttributePathParams attributePathParams2(1, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    NL_TEST_ASSERT(apSuite, attributePathParams1.IsSamePath(attributePathParams2));
+}
+
+void TestDifferentNodeId(nlTestSuite * apSuite, void * apContext)
+{
+    AttributePathParams attributePathParams1(1, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    AttributePathParams attributePathParams2(6, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    NL_TEST_ASSERT(apSuite, !attributePathParams1.IsSamePath(attributePathParams2));
+}
+
+void TestDifferentEndpointId(nlTestSuite * apSuite, void * apContext)
+{
+    AttributePathParams attributePathParams1(1, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    AttributePathParams attributePathParams2(1, 6, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    NL_TEST_ASSERT(apSuite, !attributePathParams1.IsSamePath(attributePathParams2));
+}
+
+void TestDifferentClusterId(nlTestSuite * apSuite, void * apContext)
+{
+    AttributePathParams attributePathParams1(1, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    AttributePathParams attributePathParams2(1, 2, 6, 4, 5, AttributePathFlags::kFieldIdValid);
+    NL_TEST_ASSERT(apSuite, !attributePathParams1.IsSamePath(attributePathParams2));
+}
+
+void TestDifferentFieldId(nlTestSuite * apSuite, void * apContext)
+{
+    AttributePathParams attributePathParams1(1, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    AttributePathParams attributePathParams2(1, 2, 3, 6, 5, AttributePathFlags::kFieldIdValid);
+    NL_TEST_ASSERT(apSuite, !attributePathParams1.IsSamePath(attributePathParams2));
+}
+
+void TestDifferentListIndex(nlTestSuite * apSuite, void * apContext)
+{
+    AttributePathParams attributePathParams1(1, 2, 3, 4, 5, AttributePathFlags::kListIndexValid);
+    AttributePathParams attributePathParams2(1, 2, 3, 4, 6, AttributePathFlags::kListIndexValid);
+    NL_TEST_ASSERT(apSuite, !attributePathParams1.IsSamePath(attributePathParams2));
+}
+
+void TestDifferentPathFlag(nlTestSuite * apSuite, void * apContext)
+{
+    AttributePathParams attributePathParams1(1, 2, 3, 4, 5, AttributePathFlags::kFieldIdValid);
+    AttributePathParams attributePathParams2(1, 2, 3, 4, 5, AttributePathFlags::kListIndexValid);
```

Comment:
What if `attributePathParams1` contains both `fieldId` and `listIndex` while `attributePathParams2` - only `listIndex`? Should we test such scenarios?

Reply:
if you put AttributePathParams's mFlag with 0x03, it cannot be compiled at all, it can only accept AttributePathFlags::kFieldIdValid and  AttributePathFlags::kListIndexValid